File: 02x10/c-017/main.rb
def getBritishVocal num
    case num
        when 0 then return ''
        when 1 then return 'one'
        when 2 then return 'two'
        when 3 then return 'three'
        when 4 then return 'four'
        when 5 then return 'five'
        when 6 then return 'six'
        when 7 then return 'seven'
        when 8 then return 'eight'
        when 9 then return 'nine'
        when 10 then return 'ten'
        when 11 then return 'eleven'
        when 12 then return 'twelve'
        when 13 then return 'thirteen'
        when 14 then return 'fourteen'
        when 15 then return 'fifteen'
        when 16 then return 'sixteen'
        when 17 then return 'seventeen'
        when 18 then return 'eighteen'
        when 19 then return 'nineteen'
        when 20 then return 'twenty'
        when 30 then return 'thirty'
        when 40 then return 'forty'
        when 50 then return 'fifty'
        when 60 then return 'sixty'
        when 70 then return 'seventy'
        when 80 then return 'eighty'
        when 90 then return 'ninety'
    end
    if num > 999 then
        thousandth = (num / 1000).floor
        rest = num - 1000 * thousandth
        vocalThousandth = getBritishVocal(thousandth) + ' thousand'
        vocalRest = rest > 0 ? ' and ' + getBritishVocal(rest) : ''
        return vocalThousandth + vocalRest
    end
    if num > 99 then
        hundredth = (num / 100).floor
        rest = num - 100 * hundredth
        vocalHundred = getBritishVocal(hundredth) + ' hundred'
        vocalRest = rest > 0 ? ' and ' + getBritishVocal(rest) : ''
        return vocalHundred + vocalRest
    end
    if num > 20 then
        tenth = (num / 10) * 10.floor
        onceth = num % 10
        return getBritishVocal(tenth) + '-' + getBritishVocal(onceth)
    end
end

def numberLetter
    (1..1000)
        .map { |a| getBritishVocal(a).gsub(/[ -]/, '').length }
        .sum
end

puts numberLetter

File: 02x10/c-017/main.php
<?php
function getBritishVocal($num) {
    switch ($num) {
        case 0: return '';
        case 1: return 'one';
        case 2: return 'two';
        case 3: return 'three';
        case 4: return 'four';
        case 5: return 'five';
        case 6: return 'six';
        case 7: return 'seven';
        case 8: return 'eight';
        case 9: return 'nine';
        case 10: return 'ten';
        case 11: return "eleven";
        case 12: return "twelve";
        case 13: return "thirteen";
        case 14: return "fourteen";
        case 15: return "fifteen";
        case 16: return "sixteen";
        case 17: return "seventeen";
        case 18: return "eighteen";
        case 19: return "nineteen";
        case 20: return "twenty";
        case 30: return "thirty";
        case 40: return "forty";
        case 50: return "fifty";
        case 60: return "sixty";
        case 70: return "seventy";
        case 80: return "eighty";
        case 90: return "ninety";
    }
    if ($num > 999) {
        $thousandth = floor($num / 1000);
        $rest = $num - 1000 * $thousandth;
        $vocalThousandth = getBritishVocal($thousandth) . ' thousand';
        $vocalRest = $rest > 0 ? ' and' . getBritishVocal($rest) : '';
        return $vocalThousandth . $vocalRest;
    }
    if ($num > 99) {
        $hundredth = floor($num / 100);
        $rest = $num - 100 * $hundredth;
        $vocalHundred = getBritishVocal($hundredth) . ' hundred';
        $vocalRest = $rest > 0 ? ' and' . getBritishVocal($rest) : '';
        return $vocalHundred . $vocalRest;
    }
    if ($num > 20) {
        $tenth = floor($num / 10) * 10;
        $onceth = $num % 10;
        return getBritishVocal($tenth) . '-' . getBritishVocal($onceth);
    }
}

function numberLetter() {
    global $numbers;
    $sum = 0;
    for ($i=1; $i <= 1000; $i++) {
        $sum += strlen(str_replace([" ", "-"], "", getBritishVocal($i)));
    }
    return $sum;
}

print numberLetter();
